trio/_file_io.py

```diff
@@ -154,7 +158,7 @@ def wrap_file(file):
 
     """
 
-    if isinstance(file, io.IOBase):
-        return AsyncIOWrapper(file)
+    if not hasattr(file, 'close') or not callable(file.close):
+        raise TypeError('{} does not implement required method close()'.format(file))
```

Comment:
So... the reason I suggested checking for `read` or `write` here is that it seems useful to catch mistakes where someone accidentally passes in some non-file-like object. The problem with checking for `close` is that there are lots of objects in Python that have `close` methods but aren't file-like (e.g. sockets, generators, ...), so it will tend to produce false negatives. OTOH I can't think of any case where an object can useful be called "a file" but it doesn't have `read` or `write`. (Technically I guess we don't even *require* `close` since if the user never calls our `close` method then all will be well. Though it is a bit weird that `close` shows up in `dir()` even if the wrapped object doesn't have it.) So I think checking for `read` or `write` (and possibly also `close`, I dunno) would be better?